File: javascript/easy/happyNumber.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
    let set = {};
    while(n != 1 && set[n] == null) {
        let sum = 0;
        set[n] = 1;
        while(n > 0) {
            sum += Math.pow((n%10), 2);    
            n = parseInt(n/10);
        }
        n = sum;
    }
    return (n == 1);
};

File: ruby/easy/happy_number.rb
require 'set'
# @param {Integer} n
# @return {Boolean}
def is_happy(n)
    set = Set.new
    while (n != 1 && !set.include?(n))
        sum = 0
        set.add(n)
        while n > 0
            sum += ((n % 10) ** 2)
            n /= 10
        end
        n = sum
    end
    return (n == 1)
end